src/internal/clusterstate/v2.6.0/pfsdb.go

```diff
@@ -180,179 +178,89 @@ func removeAliasCommits(ctx context.Context, tx *pachsql.Tx) error {
-		}
-		count++
-	}
-	count = 0
-	for child, parent := range childToNewParents {
-		log.Info(ctx, "updating parent after deleted alias commit",
-			zap.String("commit", oldCommitKey(child)),
-			zap.String("progress", fmt.Sprintf("%v/%v", count, len(childToNewParents))),
-		)
-		if err := updateOldCommit_V2_5(ctx, tx, child, func(ci *v2_5_0.CommitInfo) {
-			ci.ParentCommit = parent
-		}); err != nil {
-			return errors.Wrapf(err, "update child commit %q", oldCommitKey(child))
-		}
-		count++
-	}
-	// now write out realAncestors and childToNewParents
-	count = 0
-	for _, ci := range deleteCommits {
-		log.Info(ctx, "deleting alias commit",
-			zap.String("commit", oldCommitKey(ci.Commit)),
-			zap.String("progress", fmt.Sprintf("%v/%v", count, len(deleteCommits))),
-		)
-		ancstr := oldestAncestor(ci, deleteCommits)
-		// passing restoreLinks=false because we're already resetting parent/children relationships above
-		if err := deleteCommit(ctx, tx, ci.Commit, ancstr); err != nil {
-			return errors.Wrapf(err, "delete real commit %q with ancestor", oldCommitKey(ci.Commit), oldCommitKey(ancstr))
-		}
-		count++
-	}
-	// now set ci.DirectProvenance for each commit
-	if oldCIs, err = listCollectionProtos(ctx, tx, "commits", &v2_5_0.CommitInfo{}); err != nil {
-		return errors.Wrap(err, "recollect commits")
-	}
-	// re-write V2_5 commits to V2_6
-	for i, oldCI := range oldCIs {
-		var err error
-		ci := convertCommitInfoToV2_6_0(oldCI)
-		ci.DirectProvenance, err = commitProvenance(tx, ci.Commit.Repo, ci.Commit.Branch, ci.Commit.Id)
-		log.Info(ctx, "collect commit provenance",
-			zap.String("commit", oldCommitKey(ci.Commit)),
-			zap.Any("provenance", ci.DirectProvenance),
-			zap.String("progress", fmt.Sprintf("%v/%v", i, len(oldCIs))),
-		)
-		if err != nil {
-			return err
-		}
-		if err := updateCollectionProto(ctx, tx, "commits", oldCommitKey(ci.Commit), oldCommitKey(ci.Commit), ci); err != nil {
-			return errors.Wrapf(err, "update DirectProvenance for %q", oldCommitKey(ci.Commit))
 		}
+		return batcher.Close()
+	}(); err != nil {
+		return err
 	}
-	// re-point branches if necessary
-	headlessBranches := make([]*pfs.BranchInfo, 0)
-	if err := forEachCollectionProtos(ctx, tx, "branches", &pfs.BranchInfo{}, func(bi *pfs.BranchInfo) {
-		if _, ok := deleteCommits[oldCommitKey(bi.Head)]; ok {
-			headlessBranches = append(headlessBranches, proto.Clone(bi).(*pfs.BranchInfo))
-		}
-	}); err != nil {
-		return errors.Wrap(err, "record headless branches")
+	bis, err := listCollectionProtos(ctx, tx, "branches", &pfs.BranchInfo{})
+	if err != nil {
+		return err
 	}
-	for _, bi := range headlessBranches {
-		prevHead := deleteCommits[oldCommitKey(bi.Head)]
-		if err := updateOldBranch(ctx, tx, bi.Branch, func(bi *pfs.BranchInfo) {
-			bi.Head = oldestAncestor(prevHead, deleteCommits)
+	for _, bi := range bis {
+		if err := updateBranch(ctx, tx, bi.Branch, func(bi *pfs.BranchInfo) {
```

Comment:
not a correctness concern - we only need to update branches with deleted heads. Maybe this doesn't really affect performance because it's scaling with repos and not commits

Reply:
Yeah, I don't think this will have a meaningful impact on performance. I prefer the simplification and slightly lower memory footprint that comes from the removal of the delete commits map.